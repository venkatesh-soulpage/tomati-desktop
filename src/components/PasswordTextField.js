import React from "react";
// React Bootstrap
import Button from "react-bootstrap/Button";
// Font Awesome Imports
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faEye, faEyeSlash } from "@fortawesome/free-solid-svg-icons";

function PasswordTextFeild({ ...props }) {
  const [values, setValues] = React.useState({
    hidden: true
  });

  function handlePasswordToggle(event) {
    event.preventDefault();
    setValues({ ...values, hidden: !values.hidden });
  }

  return (
    <div className="input-group">
      <input
        type={values.hidden ? "password" : "text"}
        className="form-control"
        {...props}
      />
      <div className="input-group-append">
        <Button
          type="button"
          className="btn-transparent"
          onClick={handlePasswordToggle}
        >
          {values.hidden ? (
            <FontAwesomeIcon icon={faEye} className="fa-eye" />
          ) : (
            <FontAwesomeIcon icon={faEyeSlash} className="fa-eye-slash" />
          )}
        </Button>
      </div>
    </div>
  );
}

export default PasswordTextFeild;
