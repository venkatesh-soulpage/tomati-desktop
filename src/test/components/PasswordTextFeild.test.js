import React from "react";
import { render } from "@testing-library/react";
import PasswordTextField from "components/PasswordTextField";
import { Router } from "react-router-dom";
import history from "utils/history";
import { Provider } from "react-redux";
import configureStore from "store";

describe("PasswordTextField page", () => {
  it("matches snapshot", () => {
    const store = configureStore();
    const { asFragment } = render(
      <Provider store={store}>
        <Router history={history}>
          <PasswordTextField />
        </Router>
      </Provider>,
      {}
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
