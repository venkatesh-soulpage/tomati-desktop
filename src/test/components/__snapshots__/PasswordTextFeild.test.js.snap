// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordTextField page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="input-group"
  >
    <input
      class="form-control border-right-0"
      type="password"
    />
    <div
      class="input-group-append"
    >
      <div
        style="display: flex; justify-content: center; align-items: center; width: 60px; border: 1px solid #ced4da; border-top-right-radius: 5px; border-bottom-right-radius: 5px; background-color: transparent; cursor: pointer;"
      >
        <small>
          Show
        </small>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
